databricks-sdk-java/src/main/java/com/databricks/sdk/support/Paginator.java

```diff
@@ -71,6 +71,9 @@ private Iterator<T> outerIterator() {
     return new Iterator<T>() {
       @Override
       public boolean hasNext() {
+        if (currentPage == null) {
+          return false;
+        }
```

Comment:
Curious -- why wasn't this required before?

Reply:
This is kind of a separate bugfix, but I think that the query history API does not include the "results" field at all if there are no results. This is handled in flipNextPage line 63, but not in hasNext().